### ISSUE TITLE
out_file not being created by Python-function nodes

### ISSUE THREAD

**Issue by TheChymera** [nipy/nipype#2676]
Hi everybody, has Python 2.7 support been dropped? I'm getting a lot of strange errors (`out_file` not being created) on a python2.7 but not on a python 3.6 machine (granted, might be something else).

Also [the sources](https://github.com/nipy/nipype/blob/d2abc85cd98ea3a168a53306700312ca50eea87d/nipype/info.py#L59) seem to indicate python 2.7 is still supported...

**Comment by mgxd** [comment]
@TheChymera py2 is still supported, however we plan on dropping support in the 2.x release cycle.

**Comment by oesteban** [comment]
What is the error?

**Comment by TheChymera** [comment]
Ok, so the error does not seem to be Python-version-contingent, but rather happening on one specific machine. I checked and they have the same nipype, traits, prov, and pandas versions, among many other things. 

@oesteban these two nodes ([1](https://github.com/IBT-FMI/SAMRI/blob/b5abacda73e72e14b626e7b7e52a90cf8110f3b9/samri/pipelines/preprocess.py#L118),[2](https://github.com/IBT-FMI/SAMRI/blob/b5abacda73e72e14b626e7b7e52a90cf8110f3b9/samri/pipelines/preprocess.py#L121)) fail to produce an output (on one set-up out of my currently active two), which leads to errors along this line from nodes downstream:

```
TraitError: The trait 'in_file' of a SliceTimerInputSpec instance is an existing file name, but the path  '/mnt/data/ni_data/ofM.dr/preprocessing/generic_work/_ind_type_276/dummy_scans/forced_dummy_scans_file.nii.gz' does not exist.
```

**Comment by oesteban** [comment]
I think this was addressed in #2639, and rolled out with nipype-1.1.1. Which version are you using?

Could you edit the title of the issue? The fact that it is unrelated to Python's version makes it a bit misleading to other users that may have experiencing the same problem. Thanks!

**Comment by TheChymera** [comment]
@oesteban I'm on `1.1.0`, I can give the bump a try, but this doesn't really make sense --- why then would this work on another machine also using `1.1.0`?

**Comment by jdkent** [comment]
@TheChymera You may have already checked this, but I just overheard some other people having a similar error and the solution was to make sure the environment variable `FSLOUTPUTTYPE` was set to `NIFTI_GZ` in [the environment](https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FslEnvironmentVariables).

**Comment by TheChymera** [comment]
@jdkent thanks for the input, but none of the afore linked fucntions use FSL. In any case:

```
chymera@clusterhost ~ $ echo ${FSLOUTPUTTYPE}
NIFTI_GZ
```

**Comment by mgxd** [comment]
@TheChymera what OS are the two machines running?

**Comment by TheChymera** [comment]
Gentoo Linux, basically the most transparent one around. Is there anything in particular you recommend I check for?

**Comment by oesteban** [comment]
I think the question was more about whether they had different OSs. Anyways, does any of the two happen to be writing the intermediate files of the workflow under some path that is a link (or a bind mount) to somewhere else?

In more general terms, can you think of any differences between filesystems and paths between both machines?

**Comment by TheChymera** [comment]
I don't know exactly what the issue was, but at any rate it's fixed in `=nipype-1.1.1`. Thanks :)
